pkg/compiler/compiler_test.go

```diff
@@ -0,0 +1,107 @@
+package compiler
+
+import (
+	"testing"
+
+	"evylang.dev/evy/pkg/assert"
+	"evylang.dev/evy/pkg/code"
+	"evylang.dev/evy/pkg/evaluator"
+	"evylang.dev/evy/pkg/parser"
+)
+
+type compilerTestCase struct {
+	input                string
+	expectedConstants    []interface{}
+	expectedInstructions []code.Instructions
+}
+
+func TestGlobalVarStatements(t *testing.T) {
+	tests := []compilerTestCase{
+		{
+			input:             "x := 1\nx = x",
+			expectedConstants: []interface{}{1},
+			expectedInstructions: []code.Instructions{
+				code.Make(code.OpConstant, 0),
+				code.Make(code.OpSetGlobal, 0),
+				code.Make(code.OpGetGlobal, 0),
+				code.Make(code.OpSetGlobal, 1),
+			},
+		},
+		{
+			input:             "x := 2\nx = x",
+			expectedConstants: []interface{}{2},
+			expectedInstructions: []code.Instructions{
+				code.Make(code.OpConstant, 0),
+				code.Make(code.OpSetGlobal, 0),
+				code.Make(code.OpGetGlobal, 0),
+				code.Make(code.OpSetGlobal, 1),
+			},
+		},
+	}
+	runCompilerTests(t, tests)
+}
+
+func runCompilerTests(t *testing.T, tests []compilerTestCase) {
+	t.Helper()
```

Comment:
should probably not be a t.Helper() - we want to know which assertion fails.

Reply:
I agree - but the `thelper` linter made me add `t.Helper()` to every private test function that used `t *testing.T` - I think this is not a good rule because it removes the ability to have complex `assertX` test functions. From what I understand an appropriate place for `t.Helper()` statement would be somewhere like the `run` function in `evaluator_test.go` if it used a testing.T to assert failures.

Reply:
I'm not sure I agree. IMO a `TestSomething` function should do _all_ the testing without calling any other functions, other than "simple" helpers that conceptually do a "single" assertion. IMO you want to know where exactly in your `*_test.go` file an assertion fails, without having to walk a call-stack within the test file. I don't recall I've ever been linter-forced to place a `t.Helper()` where it didn't make sense to me. 

WRT to the `runVMTest` function - it is currently only called in one place, I'd inline it there. If there are more tests to come that'd call the same function, I'd still inline it everywhere despite the duplication. I guess if these tests are really only about the VM you could have a `bytecode, err := compile(prog)` function that returns any error (parse or compile) over which you run assertNoError; or you do `bytecode := mustCompile(prog)` if these steps should really never fail. Not sure if this is reasonable. I guess at this stage we are in the realm of taste.
WDYT @camh- ?